charmcraft.yaml

```diff
@@ -0,0 +1,467 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# This file configures Charmcraft.
+# See https://juju.is/docs/sdk/charmcraft-config for guidance.
+
+name: airbyte-k8s
+type: charm
+title: Airbyte Server
+summary: Airbyte Server operator
+description: |
+  Airbyte is an open-source data integration platform designed to centralize and 
+  streamline the process of extracting and loading data from various sources into 
+  data warehouses, lakes, or other destinations.
+
+links:
+  documentation: https://discourse.charmhub.io/t/charmed-airbyte-k8s-overview/14530
+
+# (Required for 'charm' type)
+bases:
+  - build-on:
+    - name: ubuntu
+      channel: "22.04"
+    run-on:
+    - name: ubuntu
+      channel: "22.04"
+
+
+# Metadata
+peers:
+  peer:
+    interface: airbyte
+
+requires:
+  db:
+    interface: postgresql_client
+    limit: 1
+
+  object-storage:
+    interface: object-storage
+    schema:
+      v1:
+        provides:
+          type: object
+          properties:
+            access-key:
+              type: string
+            namespace:
+              type:
+              - string
+              - 'null'
+            port:
+              type: number
+            secret-key:
+              type: string
+            secure:
+              type: boolean
+            service:
+              type: string
+          required:
+          - access-key
+          - port
+          - secret-key
+          - secure
+          - service
+    versions: [v1]
+    __schema_source: https://raw.githubusercontent.com/canonical/operator-schemas/master/object-storage.yaml
+
+  s3-parameters:
+    interface: s3
+    limit: 1
+    optional: true
+
+provides:
+  airbyte-server:
+    interface: airbyte-server
```

Comment:
Discussion on naming style: https://matrix.to/#/!xdClnUGkurzjxqiQcN:ubuntu.com/$oWHOH6C6yg_Ni0XpIq8X7zgGfLDgas9BEcie6xBIyMY?via=ubuntu.com&via=matrix.org&via=matrix.debian.social